### ISSUE TITLE
Path issue with ansible_local on Windows

### ISSUE THREAD

**Issue by megheaiulian** [mitchellh/vagrant#6757]
I have this in my Vagrantfile:
```ruby
#Multimachine: the develop machine
	config.vm.define :develop, primary: true do |dev|
		dev.vm.provider :virtualbox
		dev.vm.hostname = "#{project}.local"
		dev.vm.network :private_network, ip: "192.168.100.111"
		dev.vm.provision :ansible_local do |ansible|
			ansible.playbook = "provision/ansible/develop.yml"
			ansible.extra_vars = {project: project}
		end
	end
```
When I run provision the path is incorrectly transformed to E:/vagrant/provision/ansible/develop.yml:
https://i.imgur.com/vpxn4jV.png

**Comment by gildegoma** [comment]
@megheaiulian closing this one, as this actually a dupe of #6740.

This annoying bug will be fixed in 1.8.2, and in the meantime you can manually apply the patch in your Vagrant installation (07f3d0b00dabc37281a01c6776eed22daeea7066).

**Comment by megheaiulian** [comment]
@gildegoma The same problem happens with ansible.galaxy_role_file even after I apply that patch.

**Comment by gildegoma** [comment]
Oh, good catch! Indeed, in this case the absolute path is not only computed for "existence check", but also to build the `ansible-galaxy` command.

@megheaiulian Do you want to provide a PR to fix it, or should I assign myself?

**Comment by gildegoma** [comment]
@sethvargo @mitchellh do you agree to create a little shared helper function to address "Windows path handling" in a unified (and hopefully safer) way (see #6741, #542) ?

**Comment by megheaiulian** [comment]
@gildegoma I think someone else should do this because i am not familiar enough with the Vagrant codebase & structure.

**Comment by gildegoma** [comment]
@megheaiulian thanks for quick feedback, I'll try to fix it asap, but would greatly appreciate your help for pre-release testing ;-)

**Comment by megheaiulian** [comment]
@gildegoma No problem. I'll test it.

**Comment by MooazHammam** [comment]
@gildegoma 
I fixed the path issue as applying the mention patch 
for the ansible_galaxy I edited the provisioner/base.rb
as follow
line 182
def get_galaxy_role_file(basedir)
         File.expand_path(config.galaxy_role_file, basedir).gsub(/^[a-zA-Z]:/, "")
        end

        def get_galaxy_roles_path(basedir)
          if config.galaxy_roles_path
            File.expand_path(config.galaxy_roles_path, basedir).gsub(/^[a-zA-Z]:/, "")
          else
            File.join(Pathname.new(config.playbook).expand_path(basedir).parent, 'roles').gsub(/^[a-zA-Z]:/, "")
          end
        end

**Comment by gildegoma** [comment]
I've pushed a fix in #6800, but I haven't had time to run any QA on a Windows box. Help welcome :)

**Comment by megheaiulian** [comment]
@gildegoma I have added the changes from this pull request to C:\HashiCorp\Vagrant\embedded\gems\gems\vagrant-1.8.1\ .
Vagrant is now downloading galaxy roles and provisioning the VM correctly.
Thank you!

**Comment by gildegoma** [comment]
@megheaiulian :heart: :heart: :heart: thanks for your quick feedback :+1:

**Comment by gildegoma** [comment]
This issue will be closed after #6800 merge (planned for next bug fix release - 1.8.2).

More test feedbacks are still welcome until then.
